### ISSUE TITLE
certbot error during test local

### ISSUE THREAD

**Issue by mhf-ir** [letsencrypt/pebble#280]
I'v got an error:

### The request message was malformed :: Challenge response body contained legacy KeyAuthorization field, POST body should be `{}`

I run pebble :
```
./pebble  -config config.json 
Pebble 2019/10/01 00:17:31 Starting Pebble ACME server
Pebble 2019/10/01 00:17:33 Generated new root issuer with serial 61b4d6fa90e45184
Pebble 2019/10/01 00:17:33 Generated new intermediate issuer with serial 69966c968cab2db7
Pebble 2019/10/01 00:17:33 Using system DNS resolver for ACME challenges
Pebble 2019/10/01 00:17:33 Disabling random VA sleeps
Pebble 2019/10/01 00:17:33 Disabling VA challenge requests. VA always returns valid
Pebble 2019/10/01 00:17:33 Configured to reject 5% of good nonces
Pebble 2019/10/01 00:17:33 Configured to attempt authz reuse for each identifier 50% of the time
Pebble 2019/10/01 00:17:33 Management interface listening on: 0.0.0.0:15000
Pebble 2019/10/01 00:17:33 Root CA certificate available at: https://0.0.0.0:15000/roots/0
Pebble 2019/10/01 00:17:33 Listening on: 0.0.0.0:14000
Pebble 2019/10/01 00:17:33 ACME directory available at: https://0.0.0.0:14000/dir
```
Config file
```json
{
  "pebble": {
    "listenAddress": "0.0.0.0:14000",
    "managementListenAddress": "0.0.0.0:15000",
    "certificate": "cert.pem",
    "privateKey": "key.pem",
    "httpPort": 5002,
    "tlsPort": 5001,
    "ocspResponderURL": ""
  }
}
```
Install certbot then run for testing domain:

```
REQUESTS_CA_BUNDLE=pebble.minica.pem certbot certonly --manual --preferred-challenges dns --email samplemail@gmail.com --agree-tos -d jocker.sampltld --debug-challenge --server https://127.0.0.1:14000/dir
Saving debug log to /var/log/letsencrypt/letsencrypt.log
Plugins selected: Authenticator manual, Installer None
Obtaining a new certificate
Performing the following challenges:
dns-01 challenge for jocker.sampltld

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
NOTE: The IP of this machine will be publicly logged as having requested this
certificate. If you're running certbot in manual mode on a machine that is not
your server, please ensure you're okay with that.

Are you OK with your IP being logged?
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
(Y)es/(N)o: yes

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Please deploy a DNS TXT record under the name
_acme-challenge.jocker.sampltld with the following value:

LJAjALPP1M7fLlxPR5go7j-gfgiWRU1kyahV5NFFaOI

Before continuing, verify the record is deployed.
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Press Enter to Continue
Waiting for verification...

- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Challenges loaded. Press continue to submit to CA. Pass "-v" for more info about
challenges.
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
Press Enter to Continue-v
Cleaning up challenges
An unexpected error occurred:
The request message was malformed :: Challenge response body contained legacy KeyAuthorization field, POST body should be `{}`
Please see the logfiles in /var/log/letsencrypt for more details.
```

My local dns server is response the txt record:
```
dig @127.0.0.1 txt _acme-challenge.jocker.sampltld

; <<>> DiG 9.11.3-1ubuntu1.9-Ubuntu <<>> @127.0.0.1 txt _acme-challenge.jocker.sampltld
; (1 server found)
;; global options: +cmd
;; Got answer:
;; ->>HEADER<<- opcode: QUERY, status: NOERROR, id: 57569
;; flags: qr aa rd ra ad; QUERY: 1, ANSWER: 1, AUTHORITY: 0, ADDITIONAL: 1

;; OPT PSEUDOSECTION:
; EDNS: version: 0, flags:; udp: 4096
;; QUESTION SECTION:
;_acme-challenge.jocker.sampltld.	IN	TXT

;; ANSWER SECTION:
_acme-challenge.jocker.sampltld. 0 IN	TXT	"LJAjALPP1M7fLlxPR5go7j-gfgiWRU1kyahV5NFFaOI"

;; Query time: 0 msec
;; SERVER: 127.0.0.1#53(127.0.0.1)
;; WHEN: Tue Oct 01 00:19:09 +0330 2019
;; MSG SIZE  rcvd: 114

```
And logs
```
2019-10-01 00:19:20,459:DEBUG:certbot.error_handler:Calling registered functions
2019-10-01 00:19:20,459:INFO:certbot.auth_handler:Cleaning up challenges
2019-10-01 00:19:20,460:DEBUG:certbot.log:Exiting abnormally:
Traceback (most recent call last):
  File "/usr/bin/certbot", line 11, in <module>
    load_entry_point('certbot==0.31.0', 'console_scripts', 'certbot')()
  File "/usr/lib/python3/dist-packages/certbot/main.py", line 1365, in main
    return config.func(config, plugins)
  File "/usr/lib/python3/dist-packages/certbot/main.py", line 1250, in certonly
    lineage = _get_and_save_cert(le_client, config, domains, certname, lineage)
  File "/usr/lib/python3/dist-packages/certbot/main.py", line 121, in _get_and_save_cert
    lineage = le_client.obtain_and_enroll_certificate(domains, certname)
  File "/usr/lib/python3/dist-packages/certbot/client.py", line 410, in obtain_and_enroll_certificate
    cert, chain, key, _ = self.obtain_certificate(domains)
  File "/usr/lib/python3/dist-packages/certbot/client.py", line 353, in obtain_certificate
    orderr = self._get_order_and_authorizations(csr.data, self.config.allow_subset_of_names)
  File "/usr/lib/python3/dist-packages/certbot/client.py", line 389, in _get_order_and_authorizations
    authzr = self.auth_handler.handle_authorizations(orderr, best_effort)
  File "/usr/lib/python3/dist-packages/certbot/auth_handler.py", line 82, in handle_authorizations
    self._respond(aauthzrs, resp, best_effort)
  File "/usr/lib/python3/dist-packages/certbot/auth_handler.py", line 165, in _respond
    self._send_responses(aauthzrs, resp, chall_update)
  File "/usr/lib/python3/dist-packages/certbot/auth_handler.py", line 198, in _send_responses
    self.acme.answer_challenge(achall.challb, resp)
  File "/usr/lib/python3/dist-packages/acme/client.py", line 158, in answer_challenge
    response = self._post(challb.uri, response)
  File "/usr/lib/python3/dist-packages/acme/client.py", line 95, in _post
    return self.net.post(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/acme/client.py", line 1185, in post
    return self._post_once(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/acme/client.py", line 1202, in _post_once
    response = self._check_response(response, content_type=content_type)
  File "/usr/lib/python3/dist-packages/acme/client.py", line 1054, in _check_response
    raise messages.Error.from_json(jobj)
acme.messages.Error: urn:ietf:params:acme:error:malformed :: The request message was malformed :: Challenge response body contained legacy KeyAuthorization field, POST body should be `{}`
2019-10-01 00:19:20,460:ERROR:certbot.log:An unexpected error occurred:
2019-10-01 00:19:20,460:ERROR:certbot.log:The request message was malformed :: Challenge response body contained legacy KeyAuthorization field, POST body should be `{}`
```

What did i missed ?

**Comment by jsha** [comment]
```
2019-10-01 00:19:20,460:ERROR:certbot.log:The request message was malformed :: Challenge response body contained legacy KeyAuthorization field, POST body should be `{}`
```

Early ACME drafts specified that challenge responses should include a field "KeyAuthorization". Later ACME drafts, including the final one, removed that field. Pebble chooses to be very strict about interpreting the final ACME spec (RFC 8555), but Certbot still has some older behaviors. I think current Certbot probably isn't compatible with Pebble.

**Comment by cpu** [comment]
> I think current Certbot probably isn't compatible with Pebble

I believe this **was** fixed with https://github.com/certbot/certbot/commit/339d034d6a5a57d296607795a4706203f81d7059 

It looks to me that it was included in 0.38.0. @mhf-ir You may have better luck if you update your Certbot to 0.38.0+
